extra-builtins.cc

```diff
@@ -18,7 +18,7 @@ using namespace nix;
 
 struct ExtraBuiltinsSettings : Config {
     Setting<Path> extraBuiltinsFile{this,
-        settings.nixConfDir + "/extra-builtins.nix",
+        fmt("%s/%s", settings.nixConfDir, "extra-builtins.nix"),
```

Comment:
@dguibert @rehno-lindeque This change looks reasonable but I don't understand its relationship to the commit message.

Reply:
Unfortunately I didn't write the commit message, but just to mention a possible connection: 

Before I tackled this I was having some trouble with dynamic linking similar to #5. However, nix-plugins was v6.0.0 which dates after the fix for #5. So I couldn't understand why I was getting it:

```
$ nixos-rebuild test
building Nix...
error: could not dynamically open plugin file '/nix/store/5yj0rff5aql8ylbq55bp5ppag0iz8dfd-nix-plugins-6.0.0/lib/nix/plugins/libnix-extra-builtins.so': /nix/store/5yj0rff5aql8ylbq55bp5ppag0iz8dfd-nix-plugins-6.0.0/lib/nix/plugins/libnix-extra-builtins.so: undefined symbol: _ZN3nix11BaseSettingINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEE3setERKS6_
building the system configuration...
```

The commit message sounds related to this issue, but I haven't taken any time to understand it myself...

(I'm afraid I don't remember whether or not I'd set `nix.package = pkgs.nixFlakes` at the time, but it forced me to rollback my machine to a previous activation in order to regain `nixos-rebuild`, even `--option plugin-files ''` wouldn't fix the issue.)

Reply:
From what I remember, this modification fixes this "undefined symbol".

Reply:
I do not see this error on recent `nixFlakes` after my push closing this PR. Please open a bug with repo steps if this is still an issue.